manager/src/main/java/com/redhat/service/bridge/manager/connectors/AbstractConnectorWorker.java

```diff
@@ -0,0 +1,126 @@
+package com.redhat.service.bridge.manager.connectors;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.UUID;
+
+import javax.inject.Inject;
+import javax.persistence.EntityManager;
+import javax.transaction.Transactional;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.manager.models.ConnectorEntity;
+
+import io.vertx.mutiny.core.eventbus.EventBus;
+
+// R is the type returned by the external service
+public abstract class AbstractConnectorWorker<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractConnectorWorker.class);
+
+    @Inject
+    protected EntityManager entityManager;
+
+    @Inject
+    protected EventBus eventBus;
+
+    protected abstract R callExternalService(ConnectorEntity connectorEntity);
+
+    // Don't need to implement this if this is a deletion worker
+    protected ConnectorEntity updateEntityForSuccess(ConnectorEntity connectorEntity, R serviceResponse) {
+        return connectorEntity;
+    }
+
+    protected abstract ConnectorEntity updateEntityForError(ConnectorEntity connectorEntity, Throwable error);
+
+    protected void afterSuccessfullyUpdated(ConnectorEntity c) {
+    }
+
+    // Used for workers that should delete the entity after successfully calling the service.
+    // By default, after the service call updateEntityForSuccess is called, with this physical deletion will occur
+    protected boolean shouldDeleteAfterSuccess() {
+        return false;
+    }
+
+    // Could be the k8s pod name
+    String workerId = UUID.randomUUID().toString();
+
+    @Transactional(Transactional.TxType.NEVER)
```

Comment:
@lucamolteni I'd suggest adding a comment here as to why `TxType.NEVER` is important on this method for other people coming to the code who are perhaps less experienced with JPA/Hibernate than you.